### DISCUSSION TITLE
New breaking change? "Listings must have at least 1 image."

### DISCUSSION THREAD

**Discussion by stickfigure** [etsy/open-api#1084] [Q&A] [answered]
I'm suddenly getting this error from my CI system when deleting an image:

`400: {"error":"Listings must have at least 1 image. Please add another ListingImage before trying to delete."}`

This test was passing up until yesterday. The relevant code updates images by deleting and then adding.

Was there an unannounced breaking change? I don't see anything in the recent breaking changes notes.

**Comment by jasonpincin** [comment, marked as answer]
Hello @stickfigure and thanks for the question!

Etsy.com has long required at least one image be associated with an image. The public API was recently updated to be consistent with this requirement. I suggest reversing the logic of your operations for updating an image: add, then delete.

**Reply by stickfigure** [reply]
Unfortunately it's not a trivial change on our end (the logic is more complicated than a simple one-way sync). We'll start working on it but... c'mon, this is a significant breaking change to the API, it deserves an announcement and maybe a little lead time.

**Reply by nateswart** [reply]
We always strive to give _at least_ 30 days notice for any breaking change. Unplanned breaking changes are avoided whenever possible and was not an option for this issue, unfortunately.
